sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala

```diff
@@ -119,4 +113,56 @@ trait GpuPartitioning extends Partitioning with Arm {
       sliceRange.close()
     }
   }
+
+  /**
+   * Compress contiguous tables representing the splits into compressed columnar batches.
+   * Contiguous tables corresponding to splits with no data will not be compressed.
+   * @param outputBatches where to collect the corresponding columnar batches for the splits
+   * @param codec compression codec to use
+   * @param contiguousTables contiguous tables to compress
+   */
+  def compressSplits(
+      outputBatches: ArrayBuffer[ColumnarBatch],
+      codec: TableCompressionCodec,
+      contiguousTables: Array[ContiguousTable]): Unit = {
+    withResource(codec.createBatchCompressor(maxCompressionBatchSize,
+        Cuda.DEFAULT_STREAM)) { compressor =>
+      // tracks batches with no data and the corresponding output index for the batch
+      val emptyBatches = new ArrayBuffer[(ColumnarBatch, Int)]
+
+      // add each table either to the batch to be compressed or to the empty batch tracker
+      contiguousTables.zipWithIndex.foreach { case (ct, i) =>
+        if (ct.getTable.getRowCount == 0) {
+          withResource(ct) { _ =>
+            emptyBatches.append((GpuColumnVector.from(ct.getTable), i))
+          }
+        } else {
+          compressor.addTableToCompress(ct)
+        }
+      }
+
+      withResource(compressor.finish()) { compressedTables =>
+        var compressedTableIndex = 0
+        var outputIndex = 0
+        emptyBatches.foreach { case (emptyBatch, emptyOutputIndex) =>
```

Comment:
If the first batch we encountered in the foreach in line 134 is non-empty, I think the `require(emptyOutputIndex >= outputIndex)` below will trigger correctly so we would have caught this. That said, I think we want to start with the compressed table collection if the first empty batch is not at 0. I could be missing something.

Reply:
On second thought `require(emptyOutputIndex >= outputIndex)` will not trigger in this case I brought up.

Reply:
> I think we want to start with the compressed table collection if the first empty batch is not at 0.

That is [already happening on lines 149-155](https://github.com/NVIDIA/spark-rapids/blob/5b36b134b77c24a058804ceefd319417bf5f9c96/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala#L149-L155).  It adds any compressed batches that should appear before the empty batch then adds the empty batch after that.

Reply:
oh somehow skipped those lines. Thanks @jlowe